protobuf/model_config.proto

```diff
@@ -1382,6 +1382,36 @@ message ModelSequenceBatching
     //@@     The optional field to specify the initial state for the model.
     //@@
     repeated InitialState initial_state = 5;
+
+    //@@  .. cpp:var:: bool use_same_buffer_for_input_output
+    //@@
+    //@@     The optional field to use a single buffer for both input and output
+    //@@     state. Without this option, Triton uses separate buffers for input
+    //@@     and output state which can be problematic if the state size is
+    //@@     large. This option can help reduce the memory usage when
+    //@@     the state size is large. There is no harm in always enabling
+    //@@     this option if the output state will be written after
+    //@@     the input state is read by the framework. Note that when using this
+    //@@     option, `TRITONBACKEND_StateUpdate` has no effect and the state
+    //@@     will be always updated.
+    //@@     The default value is false.
+    //@@
+    bool use_same_buffer_for_input_output = 6;
+
+    //@@  .. cpp:var:: bool use_growable_memory
+    //@@
+    //@@     The optional field to allow an implicit state buffer to grow the
```

Comment:
Does this need to be optional? Wondering if we could always grow memory if using CUDA memory in the implementation.